packages/headless/src/features/search-parameters/search-parameter-analytics-actions.ts

```diff
@@ -0,0 +1,150 @@
+import {SearchParameters} from '../../features/search-parameters/search-parameter-actions';
+import {logSearchboxSubmit} from '../../features/query/query-analytics-actions';
+import {
+  AnalyticsType,
+  makeNoopAnalyticsAction,
+} from '../../features/analytics/analytics-utils';
+import {logResultsSort} from '../../features/sort-criteria/sort-criteria-analytics-actions';
+import {
+  logFacetClearAll,
+  logFacetDeselect,
+  logFacetSelect,
+} from '../../features/facets/facet-set/facet-set-analytics-actions';
+import {DateRangeRequest, NumericRangeRequest} from '../../controllers';
+
+export function logParametersChange(
+  previousParameters: SearchParameters,
+  newParameters: SearchParameters
+) {
+  if (previousParameters.q !== newParameters.q) {
+    return logSearchboxSubmit();
+  }
+
+  if (previousParameters.sortCriteria !== newParameters.sortCriteria) {
+    return logResultsSort();
+  }
+
+  if (areFacetParamsEqual(previousParameters.f, newParameters.f)) {
+    return logFacetAnalyticsAction(previousParameters.f, newParameters.f);
+  }
+
+  if (areFacetParamsEqual(previousParameters.cf, newParameters.cf)) {
+    return logFacetAnalyticsAction(previousParameters.cf, newParameters.cf);
+  }
+
+  if (areFacetParamsEqual(previousParameters.nf, newParameters.nf)) {
+    return logRangeFacetAnalyticsAction(
+      previousParameters.nf,
+      newParameters.nf
+    );
+  }
+
+  if (areFacetParamsEqual(previousParameters.df, newParameters.df)) {
+    return logRangeFacetAnalyticsAction(
+      previousParameters.df,
+      newParameters.df
+    );
+  }
+
+  // TODO: handle other possible parameters e.g. firstResult, numberOfResults
+
+  return makeNoopAnalyticsAction(AnalyticsType.Search)();
```

Comment:
If it gets here, it would effectively be "a bug in headless", that we'd want to know about.

Because then we'd be missing some analytics logging (after the todo's are added)

I guess we should logger.error ? 🤔